apps/condo/domains/user/components/auth/InputPhoneForm.tsx

```diff
@@ -52,6 +54,11 @@ export const InputPhoneForm: React.FC<IInputPhoneFormProps> = ({ onFinish })=> {
     }, [intl])
 
     const startConfirmPhone = useCallback(async () => {
+        console.log(token, isConfirmed)
+        if (token && isConfirmed) {
+            router.push(`/auth/confirm?token=${token}`)
+            return
+        }
```

Comment:
This code will not be executed after you call `setToken`, because it is wrapped into function, that is called only in form `onFinish`. Extract it to effect, like I mentioned below.